bip-0346.md

```diff
@@ -0,0 +1,438 @@
+```
+  BIP: 346
+  Layer: Consensus (soft fork)
+  Title: OP_TXHASH and OP_CHECKTXHASHVERIFY
+  Author: Steven Roose <steven@roose.io>, Brandon Black <freedom@reardencode.com>
+  Comments-URI: https://github.com/bitcoin/bips/wiki/Comments:BIP-0346
+  Status: Draft
+  Type: Standards Track
+  Created: 2023-09-03
+  License: BSD-3-Clause
+```
+
+# Abstract
+
+This BIP proposes two new opcodes, `OP_CHECKTXHASHVERIFY`, to be activated
+as a change to the semantics of `OP_NOP4` in legacy script, segwit and tapscript;
+and `OP_TXHASH`, to be activated as a change to the semantics of `OP_SUCCESS189`
```

Comment:
I double checked, OP_SUCCESS appears to be unclaimed, OP_VAULT used `OP_SUCCESS187` and `OP_SUCCESS188`.